src/main/java/org/opensearch/neuralsearch/processor/combination/ArithmeticMeanScoreCombinationTechnique.java

```diff
@@ -5,16 +5,34 @@
 
 package org.opensearch.neuralsearch.processor.combination;
 
-import lombok.NoArgsConstructor;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+
+import org.opensearch.OpenSearchParseException;
 
 /**
  * Abstracts combination of scores based on arithmetic mean method
  */
-@NoArgsConstructor
 public class ArithmeticMeanScoreCombinationTechnique implements ScoreCombinationTechnique {
 
     public static final String TECHNIQUE_NAME = "arithmetic_mean";
+    public static final String PARAM_NAME_WEIGHTS = "weights";
     private static final Float ZERO_SCORE = 0.0f;
+    private final List<Double> weights;
```

Comment:
our parser does doubles, those are not vectors but just factors for multiplication, so double or float doesn't really matter. Otherwise we need to do one more round of conversion, not sure if that's necessary

Reply:
Isn't this the place where we get weight? Here can we just cast it to `(List<Float>)`?
```
weights = (List<Double>) params.getOrDefault(PARAM_NAME_WEIGHTS, new ArrayList<>());
```

Reply:
cast wouldn't work. we can only convert it

Reply:
I added explicit conversion double -> float